src/utilities/attachments-manager.ts

```diff
@@ -0,0 +1,91 @@
+import * as AWS from "aws-sdk";
+import { Credentials, S3Resource, TokenServiceClient } from "@concord-consortium/token-service";
+import { firebaseAppName } from "../portal-api";
+import { IAttachmentsFolder } from "../types";
+
+const kTokenServiceToolName = "interactive-attachments";
+const kDefaultWriteExpirationSec = 5 * 60;
+const kDefaultReadExpirationSec = 2 * 60 * 60;
+
+let resolveAttachmentsManager: (manager: AttachmentsManager) => void;
+export const attachmentsManager = new Promise<AttachmentsManager>((resolve, reject) => {
+  resolveAttachmentsManager = resolve;
+});
+
+export const initializeAttachmentsManager = (learnerId: string, firebaseJwt?: string) => {
+  resolveAttachmentsManager(new AttachmentsManager(learnerId, firebaseJwt));
+};
+
+export class AttachmentsManager {
+  private learnerId?: string;
+  private firebaseJwt?: string;
+  private tokenServiceClient: TokenServiceClient;
+  private resources: Record<string, S3Resource> = {};
+
+  constructor(learnerId: string, firebaseJwt?: string) {
+    this.learnerId = learnerId;
+    this.firebaseJwt = firebaseJwt;
```

Comment:
The JWT for the token service is different than the JWT for the report service or portal.
I think the code calling this is passing the JWT for the report service here. 

What it should probably do instead if pass the portal JWT, so this variable shouldn't be called `portalJwt`. And then if it has a `portalJwt`, then the attachments manager requests a token service jwt. The token service "app name" in the staging and production portals is `token-service`.   

You can kind of trace similar code here: https://github.com/concord-consortium/report-service/blob/efce8eb9d2b8d1484c35c7c74888e3380b472799/researcher-reports/src/components/app.tsx#L45

In that case it is bit more complex because it has to use OAuth to get the portal JWT. In this case the AP should already have a portal JWT since I think it needed that for other stuff.

Reply:
The value I'm passing in is generated at https://github.com/concord-consortium/activity-player/blob/df7fe8dcc1e439c819829b478c549955e915a6d9/src/portal-api.ts#L412, which seemed like it might be appropriate.

Reply:
I've rewritten this to request a "token-service"-specific JWT from the information in the PortalData.